### PR TITLE
New guide about monitoring a validator node 

### LINKED ISSUE #1422: New guide: Monitoring a validator node
**Is your feature request related to a problem? Please describe.**
Monitoring of a delegate is particulary important for delegates forging on the Mainnet. The docs should provide a simple example of doing so.

**Describe the solution you'd like**
Add a recommended solution and example for monitoring a delegate.

### PR DESCRIPTION
**Pull request by TalhaMaliktz**
Closes #1422

### CODE REVIEW

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+:url_prometheus: https://prometheus.io/docs/introduction/overview/
+:url_monitor_endpoints: https://github.com/LiskHQ/lisk-sdk/blob/development/framework-plugins/lisk-framework-monitor-plugin/src/endpoint.ts
+
+// Project URLs
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: v4@lisk-core::index.adoc
+
+The {url_plugin_monitor}[Monitor plugin^] provides insights about a running node, for example, if a node is down or a node is missing blocks, etc.
+The Monitor plugin takes the data reported by {url_prometheus}[Prometheus (an open-source systems monitoring and alerting toolkit)^] and exposes the data into a standard format using its own <<retrieving-data ,endpoints>>.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: Does the monitor plugin really take data from prometheus, or is it the other way around?

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+:url_cors_intro: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+:url_prometheus: https://prometheus.io/docs/introduction/overview/
+:url_monitor_endpoints: https://github.com/LiskHQ/lisk-sdk/blob/development/framework-plugins/lisk-framework-monitor-plugin/src/endpoint.ts
+
+// Project URLs
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: v4@lisk-core::index.adoc
+
+The {url_plugin_monitor}[Monitor plugin^] provides insights about a running node, for example, if a node is down or a node is missing blocks, etc.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: The link doesnt provide much useful information, as we wont be updating the typedocs, so imho its better to remove the link
```suggestion
The *Monitor plugin* provides insights about a running node, for example, if a node is down or a node is missing blocks, etc.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+:url_plugin_monitor: {site-url}/lisk-sdk/v6/references/typedoc/modules/_liskhq_lisk_framework_monitor_plugin.html
+:url_plugin_monitor_config: {site-url}/lisk-sdk/v6/references/typedoc/modules/_liskhq_lisk_framework_monitor_plugin.html#$config-options
+:url_cors_intro: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+:url_prometheus: https://prometheus.io/docs/introduction/overview/
+:url_monitor_endpoints: https://github.com/LiskHQ/lisk-sdk/blob/development/framework-plugins/lisk-framework-monitor-plugin/src/endpoint.ts
+
+// Project URLs
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: v4@lisk-core::index.adoc
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
:url_core_index: lisk-core::index.adoc
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+The {url_plugin_monitor}[Monitor plugin^] provides insights about a running node, for example, if a node is down or a node is missing blocks, etc.
+The Monitor plugin takes the data reported by {url_prometheus}[Prometheus (an open-source systems monitoring and alerting toolkit)^] and exposes the data into a standard format using its own <<retrieving-data ,endpoints>>.
+
+====
+On this page, you'll learn:
+
+* [x] Registration of the Monitor plugin to a blockchain client.
+* [x] Configuring the Monitor plugin.
+* [x] Retrieving various types of data from the Monitor plugin.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: if this is the only content on this page, it should be renamed to "How to enable the Monitor plugin" or something similar, as it doesn't provide any additional approach and also no bigger picture like context and use cases for monitoring different metrics.

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 2 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: Instead, please introduce the Monitor plugin a little more directly in the intro of this guide, what are its main features and use cases?

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    "monitor": {
+        "port": "9000"
+    }
+}
+----
+--
+=====
+
+
+== Retrieving data
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc:218]: Here, I miss usage examples, how to invoke the different endpoints, and how the responses look like. It should also include a link to a guide / docs that explain how to track endpoints with prometheus

**Review by Tschakki — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 1 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: Corrected

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 2 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: Done

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 4 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: Text updated

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 5 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc:218]: Usage example added

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+// Project URLs
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: lisk-core::index.adoc
+
+The *Monitor plugin* provides insights about a running node, for example:
+
+* It provides developers with vital information on block processing and network performance.
+* It also provides information to determine machine performance relative to network activity.
+* The Monitor plugin gives hints for debugging the network, and if there is an attack on the network.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* The Monitor plugin gives hints for debugging the network, and detecting any network attacks.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+* One can also receive information about the performance of the custom module via the Monitor plugin.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* Information regarding the performance of the custom module can also be obtained through the Monitor plugin.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+The Monitor plugin monitors a node's activity and stores this data in the plugin's store.
+It then exposes the aforementioned data using the endpoints as described in the <<retrieving-data>> section.
+The Monitor plugin also exports this data in the Prometheus format for the {url_prometheus}[Prometheus toolkit (an open-source systems monitoring and alerting toolkit)^].
+
+====
+On this page, you'll learn:
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
On this page, you'll learn how to perform the following:
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+* [x] Registration of the Monitor plugin to a blockchain client.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* [x] Register the Monitor plugin to a blockchain client.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+* [x] Configuring the Monitor plugin.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* [x] Configure the Monitor plugin.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+* [x] Retrieving various types of data from the Monitor plugin.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* [x] Retrieve various types of data from the Monitor plugin.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+====
+
+Using this guide, the Monitor plugin can be registered to a xref:{url_core_index}[Lisk Core] node or any Lisk-based blockchain client.
+
+== Plugin registration
+With the latest version of Lisk Core v4 and Lisk SDK v6, the default plugins come pre-installed after a client is installed or initiated, respectively.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
With the latest version of Lisk Core v4 and Lisk SDK v6, the default plugins come pre-installed after a client is installed or initiated respectively.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+ 
+You can check the `package.json` file of the blockchain client to ensure the installation.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
You can check the `package.json` file of the blockchain client to ensure the installation has been performed correctly.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+
+Otherwise, you can install the Monitor plugin like this:
+ 
+[source,bash]
+----
+npm install --save @liskhq/lisk-framework-monitor-plugin
+----
+
+There are various ways to enable the Monitor plugin.
+You can opt for any of the below-mentioned to register the plugin to the blockchain client.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
You can opt for any of the below-mentioned options to register the plugin to the blockchain client.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+
+[IMPORTANT]
+====
+It is essential to enable WS API and configure the `allowedMethods` property before registering the `MonitorPlugin`.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
It is essential to enable the WS API and configure the `allowedMethods` property before registering the `MonitorPlugin`.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+
+After updating the *plugins.ts* file, build the blockchain client again:
+
+[source,bash]
+----
+npm run build
+----
+
+== Starting the node
+Depending on how you chose to register the plugin, start the blockchain client, accordingly.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
Depending on how you chose to register the plugin, start the blockchain client accordingly.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+[source,bash]
+----
+./bin/run start --config=config/custom_config.json 
+----
+--
+=====
+
+
+== Configuring the Monitor Plugin
+The Monitor plugin comes with a set of pre-defined configurations, which can be tweaked as per the node operator's needs.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
The Monitor plugin comes with a set of pre-defined configurations, which can be tweaked as per the node operator's requirements.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+|`127.0.0.1`
+
+|`whiteList`
+|string
+|When given a value, only those IPs are allowed to interact with the Monitor plugin.
+|`['127.0.0.1']`
+
+|`cors`
+|string
+|Defines the {url_cors_intro}[CORS^] properties such as `origin` and allowed `methods` which should be entertained by the Monitor plugin.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: imho we should just slightly reword this line to the following:
Defines the {url_cors_intro}[CORS^] properties such as `origin` and allowed `methods` which should be handled by the Monitor plugin.

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+--
+=====
+
+
+== Retrieving data
+The data recorded by the Monitor plugin can be retrieved via RPC endpoints or in Prometheus format, as described in the following sub-sections.
+
+=== Via RPC endpoints
+The monitor plugin exposes four endpoints that return important data about a validator's node.
+The following table briefly describes each:
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
The following table briefly describes them:
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+
+# HELP lisk_fork_events_total Fork events
+# TYPE lisk_fork_events_total gauge
+lisk_fork_events_total 0
+----
+
+You can configure Prometheus to automatically invoke the aforementioned endpoint after regular intervals.
+For more information, see the {url_prometheus_configuration}[Configuration^] section of the Prometheus documentation.
+
+Configuring the Prometheus to automatically invoke the aforementioned endpoint and then, plugging such data to visualizing tools like Grafana can enable a node operator to stay up to date with the latest status of their node.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
Configuring the Prometheus to automatically invoke the aforementioned endpoint and then plugging such data into visualizing tools such as Grafana, can enable a node operator to stay up to date with the latest status of their node.
```

**Review by chris529 — COMMENTED** [review]
Looks great! I just left a few minor suggestions/recommendations 👍

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 18 above)
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
|Defines the {url_cors_intro}[CORS^] properties such as origin and allowed methods which should be handled by the Monitor plugin.
```

**Review by chris529 — APPROVED** [review]
LGTM! 👍

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: lisk-core::index.adoc
+
+The *Monitor plugin* provides insights about a running node, for example:
+
+* It provides developers with vital information on block processing and network performance.
+* It also provides information to determine machine performance relative to network activity.
+* The Monitor plugin gives hints for debugging the network, and detecting any network attacks.
+* Information regarding the performance of the custom module can also be obtained through the Monitor plugin.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: I'm not sure if we can get any insight around performance of custom module

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 22]
```diff
@@ -0,0 +1,298 @@
...
+:url_prometheus_configuration: https://prometheus.io/docs/prometheus/latest/configuration/configuration/
+
+// Project URLs
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_core_index: lisk-core::index.adoc
+
+The *Monitor plugin* provides insights about a running node, for example:
+
+* It provides developers with vital information on block processing and network performance.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* It provides information about blocks and transactions received from the network.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 23]
```diff
@@ -0,0 +1,298 @@
...
+* It also provides information to determine machine performance relative to network activity.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: It does not have any information about machine performance but more like network activity on your node
```suggestion
* It provides information about connected/disconnected/banned peers and includes data about other network activities.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 24]
```diff
@@ -0,0 +1,298 @@
...
+* The Monitor plugin gives hints for debugging the network, and detecting any network attacks.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
* The Monitor plugin gives hints for debugging the network, and detecting any network splits or attacks.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 25]
```diff
@@ -0,0 +1,298 @@
...
+|Description
+
+|*monitor_getTransactionStats*
+|Returns the data about the number of times a transaction is received on an average from the network for a given number of connected peers.
+
+|*monitor_getBlockStats*
+|Returns the data about the number of times a block is received on an average from the network for a given number of connected peers.
+
+|*monitor_getNetworkStats*
+|Returns the data about the number of connected/disconnected peers, and the number of outgoing/incoming connections with several peers at a certain height.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]:
```suggestion
|Returns the data about the number of connected/disconnected/banned peers, and the number of outgoing/incoming connections with several peers at a certain height.
```

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc [hunk 26]
```diff
@@ -0,0 +1,298 @@
...
+----
+--
+=====
+
+
+== Retrieving data
+The data recorded by the Monitor plugin can be retrieved via RPC endpoints or in Prometheus format, as described in the following sub-sections.
+
+=== Via RPC endpoints
+The monitor plugin exposes four endpoints that return important data about a validator's node.
```
Comment (ishantiw) [inline docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: I'm not sure if it really helps validator node with any information around forging a block but yes in general it does help a node to see how it is connected and receiving information with the network

**Review by ishantiw — CHANGES_REQUESTED** [review]
There are few comments around functionality of monitor plugin

File: docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc (hunk 26 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/monitoring-node.adoc]: I have removed the term `validator`.

**Review by ishantiw — APPROVED** [review]
Great job @TalhaMaliktz 🙌

**Review by Tschakki — APPROVED** [review]

**Pull request merged**